test/e2e/automigration/automigration.go

```diff
@@ -228,4 +238,343 @@ var _ = ginkgo.Describe("auto migration", autoMigrationTestLabel, func() {
 			}
 		})
 	})
+
+	ginkgo.It("Should migrate unschedulable pods properly with overlapping label selectors", func(ctx context.Context) {
+		// 1. Get test clusters
+
+		sort.Slice(clusters, func(i, j int) bool {
+			return clusters[i].Name < clusters[j].Name
+		})
+		clusterToMigrateFromForDp1 := clusters[0]
+		clusterToMigrateFromForDp2 := clusters[2]
+
+		var err error
+
+		// 2. Initialize replicas
+
+		replicasPerCluster := int32(3)
+		totalReplicas := int32(len(clusters)) * replicasPerCluster
+
+		// 3. Initialize deployments
+
+		dp1 := resources.GetSimpleDeployment(f.Name())
+		dp1.Spec.Replicas = pointer.Int32(totalReplicas)
+		maxSurge := intstr.FromInt(0)
+		dp1.Spec.Strategy = appsv1.DeploymentStrategy{
+			Type: appsv1.RollingUpdateDeploymentStrategyType,
+			RollingUpdate: &appsv1.RollingUpdateDeployment{
+				// Prevent upgrade getting stuck
+				MaxSurge: &maxSurge,
+			},
+		}
+		dp2 := dp1.DeepCopy()
+
+		// 4. Create policies
+
+		propPolicy.Spec.AutoMigration.KeepUnschedulableReplicas = true
+		ginkgo.By("Creating PropagationPolicy", func() {
+			propPolicy, err = f.HostFedClient().CoreV1alpha1().PropagationPolicies(f.TestNamespace().Name).Create(
+				ctx,
+				propPolicy,
+				metav1.CreateOptions{},
+			)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred(), framework.MessageUnexpectedError)
+		})
+
+		// This function creates an override policy that results in only a single pod of a deployment being schedulable
```

Comment:
Can't we create dp2 directly in one of the member clusters, and test whether dp1 is migrated unnecessarily?